pkg/apiserver/user/auth.go

```diff
@@ -30,6 +31,8 @@ var (
 )
 
 type AuthService struct {
+	FeatureFlagNonRootLogin *featureflag.FeatureFlag
```

Comment:
This feature should be placed inside SQL auth, as only SQL auth needs it and other kind of auth does not care about the "non root login".